napari/layers/_shapes_layer/model.py

```diff
@@ -1251,7 +1348,141 @@ def to_xml_list(self, shape_type=None):
             List of xml elements defining each shape according to the
             svg specification
         """
-        return self.data.to_xml_list(shape_type=shape_type)
+        return self.slice_data.to_xml_list(shape_type=shape_type)
+
+    def to_masks(self, mask_shape=None, shape_type=None):
```

Comment:
Is that because of the special casing for the `dims=2`? I'm sure we could remove that, but I'd really much rather see the whole function go in now and we refactor with a followup PR then have to pull it out and maintain it somewhere else in the interim